modules/services/git-workspace.nix

```diff
@@ -0,0 +1,153 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.git-workspace;
+  tomlFormat = pkgs.formats.toml { };
+  workspaceType = lib.types.submodule {
+    freeformType = tomlFormat.type;
+    options.provider = lib.mkOption {
```

Comment:
All the `lib.`s are a bit overwhelming. If you don't want to use `with lib` then can put something like

``` nix
inherit (lib) mkOption literalExpression …;
```

in the top-level `let` bindings.

Reply:
Doing that, but only for things that appear multiple times.

Reply:
Doing that, but not with single types.